# FAKE REVIEW CLASSIFICATION

In [1]:
# importing all necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import string
import os
from sklearn import preprocessing, model_selection
from IPython.display import display, HTML
import matplotlib.pyplot as plt
from sklearn.decomposition import PCA

sns.set()

%matplotlib inline

In [2]:
#importing libraries
import numpy as np
import pandas as pd
import nltk
import matplotlib.pyplot as plt
import statistics
import math
import os
from sklearn.model_selection import KFold, StratifiedKFold, cross_val_score
from sklearn.metrics import confusion_matrix
nltk.download('stopwords')
nltk.download('punkt')
nltk.download('wordnet')
nltk.download('omw-1.4')

import re
pd.set_option('max_colwidth', 800)


#importing Libraies
import networkx as nx
import matplotlib.pyplot as plt
from numpy import genfromtxt
import collections
from collections import OrderedDict

import matplotlib.pyplot as plt
%matplotlib inline
from sklearn.model_selection import train_test_split
from sklearn.model_selection import KFold
from sklearn.model_selection import cross_val_score
from sklearn.tree import DecisionTreeClassifier
from sklearn.feature_selection import SelectFromModel
from sklearn.svm import SVC
from sklearn.preprocessing import StandardScaler,MinMaxScaler
from sklearn.linear_model import LinearRegression,LogisticRegression, RidgeClassifier
from sklearn.discriminant_analysis import LinearDiscriminantAnalysis, QuadraticDiscriminantAnalysis
from sklearn.ensemble import RandomForestClassifier
from sklearn.preprocessing import LabelEncoder
from sklearn.metrics import classification_report,accuracy_score

[nltk_data] Downloading package stopwords to /root/nltk_data...
[nltk_data]   Unzipping corpora/stopwords.zip.
[nltk_data] Downloading package punkt to /root/nltk_data...
[nltk_data]   Unzipping tokenizers/punkt.zip.
[nltk_data] Downloading package wordnet to /root/nltk_data...
[nltk_data] Downloading package omw-1.4 to /root/nltk_data...


In [3]:
#loading dataset
df=pd.read_csv('/content/deceptive.csv')
df.head()

deceptive   hotel  polarity       source  \
0  truthful  conrad  positive  TripAdvisor   
1  truthful   hyatt  positive  TripAdvisor   
2  truthful   hyatt  positive  TripAdvisor   
3  truthful    omni  positive  TripAdvisor   
4  truthful   hyatt  positive  TripAdvisor   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              text  \
0                                                                                                                                                                                                                                    We stayed for a one night getaway with family on a thursday. Triple AAA rate of 173 was a steal. 7th floor room complete with 44in plasma TV bose stereo, voss and evian water, and gorgeous bathroom(no tub but was fine for us) Concierge was very helpful. You cannot beat this location... Only flaw was breakfast was pricey and service was very very slow(2hours for four kids and four adults on a friday morning) even though there were only two other tables in the restaurant. Food was very good so it was worth the wait. I would return in a heartbeat. A gem in chicago... \n   
1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Triple A rate with upgrade to view room was less than $200 which also included breakfast vouchers. Had a great view of river, lake, Wrigley Bldg. & Tribune Bldg. Most major restaurants, Shopping, Sightseeing attractions within walking distance. Large room with a very comfortable bed. \n   
2  This comes a little late as I'm finally catching up on my reviews from the past several months:) A dear friend and I stayed at the Hyatt Regency in late October 2007 for one night while visiting a friend and her husband from out of town. This hotel is perfect, IMO. Easy check in and check out. Lovely, clean, comfortable rooms with great views of the city. I know this area pretty well and it's very convenient to many downtown Chicago attractions. We had dinner and went clubing with our friends around Division St.. We had no problems getting cabs back and forth to the Hyatt and there's even public transportation right near by but we didn't bother since we only needed cabs from and to the hotel. Parking, as is usual for Chicago, was expensive but we were able to get our car out quickly (h...   
3                                                                                             The Omni Chicago really delivers on all fronts, from the spaciousness of the rooms to the helpful staff to the prized location on Michigan Avenue. While this address in Chicago requires a high level of quality, the Omni delivers. Check in for myself and a whole group of people with me was under 3 minutes, the staff had plentiful recommendations for dining and events, and the rooms are some of the largest

In [4]:
d = []
for i in range(1600):
    s = i+1
    d.append(s)

In [5]:
#d

In [6]:
df.insert(0, 'Index', d)
df.head()

Index deceptive   hotel  polarity       source  \
0      1  truthful  conrad  positive  TripAdvisor   
1      2  truthful   hyatt  positive  TripAdvisor   
2      3  truthful   hyatt  positive  TripAdvisor   
3      4  truthful    omni  positive  TripAdvisor   
4      5  truthful   hyatt  positive  TripAdvisor   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              text  \
0                                                                                                                                                                                                                                    We stayed for a one night getaway with family on a thursday. Triple AAA rate of 173 was a steal. 7th floor room complete with 44in plasma TV bose stereo, voss and evian water, and gorgeous bathroom(no tub but was fine for us) Concierge was very helpful. You cannot beat this location... Only flaw was breakfast was pricey and service was very very slow(2hours for four kids and four adults on a friday morning) even though there were only two other tables in the restaurant. Food was very good so it was worth the wait. I would return in a heartbeat. A gem in chicago... \n   
1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Triple A rate with upgrade to view room was less than $200 which also included breakfast vouchers. Had a great view of river, lake, Wrigley Bldg. & Tribune Bldg. Most major restaurants, Shopping, Sightseeing attractions within walking distance. Large room with a very comfortable bed. \n   
2  This comes a little late as I'm finally catching up on my reviews from the past several months:) A dear friend and I stayed at the Hyatt Regency in late October 2007 for one night while visiting a friend and her husband from out of town. This hotel is perfect, IMO. Easy check in and check out. Lovely, clean, comfortable rooms with great views of the city. I know this area pretty well and it's very convenient to many downtown Chicago attractions. We had dinner and went clubing with our friends around Division St.. We had no problems getting cabs back and forth to the Hyatt and there's even public transportation right near by but we didn't bother since we only needed cabs from and to the hotel. Parking, as is usual for Chicago, was expensive but we were able to get our car out quickly (h...   
3                                                                                             The Omni Chicago really delivers on all fronts, from the spaciousness of the rooms to the helpful staff to the prized location on Michigan Avenue. While this address in Chicago requires a high level of quality, the Omni delivers. Check in for myself and a whole group of people with me was under 3 minutes, the staff had plentiful recommendations for dining and even

In [7]:
df.shape

(1600, 7)

In [8]:
df['Class'].value_counts()

1    800
0    800
Name: Class, dtype: int64

In [9]:
df = df.sample(frac=1).reset_index(drop=True)
df.head()

Index  deceptive             hotel  polarity       source  \
0     99   truthful          sheraton  positive  TripAdvisor   
1    185   truthful          hardrock  positive  TripAdvisor   
2   1574  deceptive  intercontinental  negative        MTurk   
3    523  deceptive         swissotel  positive        MTurk   
4   1548  deceptive            palmer  negative        MTurk   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              text  \
0  We stayed at the Sheraton by Navy Pier the first weekend of November and were very pleased with the accommodations and the service. We were upgraded to club level rooms and stayed in 3114 and 3115 adjoining. The view from both rooms was spectacular (as you can tell from the picture attached). They also left a plate of cookies and treats in the kids room upon check-in which made us all feel very special. The hotel is central to both Navy Pier and Michigan Ave. so we walked, trolleyed, and cabbed all around the area. We ate the breakfast buffet both mornings and thought it was pretty good. The eggs were a little runny, but they happily made a different batch for us at the omelet station. Our six year old ate free and our two eleven year olds were $14 (instead of the adult $20) The rooms ...   
1  Got a great deal through Hotwire for $70 a night but if I ever return to Chicago I would definitely stay again, even at full price. We really liked the rooms muted modern decor, we had John Lennon and Bono for company. Comfortable bed, mini coffe maker, plenty storage space and a lovely, clean bathroom stocked full of Aveda products (so I was sold straight away...it doesn't take much, I know!). If you're a product junkie, tip the maid and request some extra bottles and you will leave with lots of goodies :) We liked the location and had a great view down N Michigan Ave, found the staff to be really friendly and relaxed (reception, gift shop and bar). The bar area is very trendy, I liked the chilled atmosphere which surprised me cause I normally find these types of bars a bit too preten...   
2  My wife and I were excited to visit and shop at Chicago's Magnificent Mile on Michigan Ave. We chose the Intercontinental Hotel for it's reputation and location. Man oh man, were we disappointed. The check in was smooth enough even though the hotel was busy and all seemed well. We got ready to go out shopping and went to visit the concierge in the lobby to ask about suggestions and bargains. There were six people in line for her and we had to wait. That was no problem since we weren't on a timetable. When we finally got to her and asked about shopping on the Miracle Mile, she just pointed to the door and said, 'It's right out the door.' I asked her for suggestions of the best stores for bargains and she rolled her eyes at me saying, 'there's no such thing.' I couldn't believe her abrup...   
3                                                                                                                                                                                                                                                                                                                                                                                                         

### Data pre-processing

In [10]:
# dropping unnecessary columns. 
df.drop(['deceptive', 'hotel', 'polarity', 'source'], axis = 1, inplace=True)
df.head()

Index  \
0     99   
1    185   
2   1574   
3    523   
4   1548   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              text  \
0  We stayed at the Sheraton by Navy Pier the first weekend of November and were very pleased with the accommodations and the service. We were upgraded to club level rooms and stayed in 3114 and 3115 adjoining. The view from both rooms was spectacular (as you can tell from the picture attached). They also left a plate of cookies and treats in the kids room upon check-in which made us all feel very special. The hotel is central to both Navy Pier and Michigan Ave. so we walked, trolleyed, and cabbed all around the area. We ate the breakfast buffet both mornings and thought it was pretty good. The eggs were a little runny, but they happily made a different batch for us at the omelet station. Our six year old ate free and our two eleven year olds were $14 (instead of the adult $20) The rooms ...   
1  Got a great deal through Hotwire for $70 a night but if I ever return to Chicago I would definitely stay again, even at full price. We really liked the rooms muted modern decor, we had John Lennon and Bono for company. Comfortable bed, mini coffe maker, plenty storage space and a lovely, clean bathroom stocked full of Aveda products (so I was sold straight away...it doesn't take much, I know!). If you're a product junkie, tip the maid and request some extra bottles and you will leave with lots of goodies :) We liked the location and had a great view down N Michigan Ave, found the staff to be really friendly and relaxed (reception, gift shop and bar). The bar area is very trendy, I liked the chilled atmosphere which surprised me cause I normally find these types of bars a bit too preten...   
2  My wife and I were excited to visit and shop at Chicago's Magnificent Mile on Michigan Ave. We chose the Intercontinental Hotel for it's reputation and location. Man oh man, were we disappointed. The check in was smooth enough even though the hotel was busy and all seemed well. We got ready to go out shopping and went to visit the concierge in the lobby to ask about suggestions and bargains. There were six people in line for her and we had to wait. That was no problem since we weren't on a timetable. When we finally got to her and asked about shopping on the Miracle Mile, she just pointed to the door and said, 'It's right out the door.' I asked her for suggestions of the best stores for bargains and she rolled her eyes at me saying, 'there's no such thing.' I couldn't believe her abrup...   
3                                                                                                                                                                                                                                                                                                                                                                                                           I must say, this hotel is absolutely beautiful. My wife and I were blown away by the penthouse fitness center and pool all the way up on the 42nd floor. Very well designed rooms with great linens and pillows. The bathrooms were a nice change of pace from the boring hotel bathrooms we usually see. Overall an e

In [11]:
# Cleaning Raw tweets
def clean_text(text):
    
    #remove emails
    text = ' '.join([i for i in text.split() if '@' not in i])
    
    #remove web address
    text = re.sub('http[s]?://\S+', '', text)

    #remove brackets and parantheses (and removing the text inside the brackets and parantheses)
    text = re.sub("[\(\[].*?[\)\]]", "", text)

    #remove punctuations
    text = re.sub(r'[^\w\s]', ' ', text)
    
    #Filter to allow only alphabets and numbers
    text = re.sub(r'[^a-zA-Z\']', ' ', text)
    
    #Remove Unicode characters
    text = re.sub(r'[^\x00-\x7F]+', '', text)

    #remove double spaces 
    text = re.sub('\s+', ' ', text)
    
    return text

In [12]:
# applying that into function.
df["clean_text"] = df['text'].apply(lambda x: clean_text(x))
df.head(10)

Index  \
0     99   
1    185   
2   1574   
3    523   
4   1548   
5    427   
6   1511   
7     11   
8   1421   
9    532   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              text  \
0  We stayed at the Sheraton by Navy Pier the first weekend of November and were very pleased with the accommodations and the service. We were upgraded to club level rooms and stayed in 3114 and 3115 adjoining. The view from both rooms was spectacular (as you can tell from the picture attached). They also left a plate of cookies and treats in the kids room upon check-in which made us all feel very special. The hotel is central to both Navy Pier and Michigan Ave. so we walked, trolleyed, and cabbed all around the area. We ate the breakfast buffet both mornings and thought it was pretty good. The eggs were a little runny, but they happily made a different batch for us at the omelet station. Our six year old ate free and our two eleven year olds were $14 (instead of the adult $20) The rooms ...   
1  Got a great deal through Hotwire for $70 a night but if I ever return to Chicago I would definitely stay again, even at full price. We really liked the rooms muted modern decor, we had John Lennon and Bono for company. Comfortable bed, mini coffe maker, plenty storage space and a lovely, clean bathroom stocked full of Aveda products (so I was sold straight away...it doesn't take much, I know!). If you're a product junkie, tip the maid and request some extra bottles and you will leave with lots of goodies :) We liked the location and had a great view down N Michigan Ave, found the staff to be really friendly and relaxed (reception, gift shop and bar). The bar area is very trendy, I liked the chilled atmosphere which surprised me cause I normally find these types of bars a bit too preten...   
2  My wife and I were excited to visit and shop at Chicago's Magnificent Mile on Michigan Ave. We chose the Intercontinental Hotel for it's reputation and location. Man oh man, were we disappointed. The check in was smooth enough even though the hotel was busy and all seemed well. We got ready to go out shopping and went to visit the concierge in the lobby to ask about suggestions and bargains. There were six people in line for her and we had to wait. That was no problem since we weren't on a timetable. When we finally got to her and asked about shopping on the Miracle Mile, she just pointed to the door and said, 'It's right out the door.' I asked her for suggestions of the best stores for bargains and she rolled her eyes at me saying, 'there's no such thing.' I couldn't believe her abrup...   
3                                                                                                                                                                                                                                                                                                                                                                                                           I must say, this hotel is absolutely beautiful. My wife and I were blown away by the penthouse fitness center and pool all the way up on the 42nd floor. Very well designed rooms with great linens and pillows. The bathrooms were a nice change of pace 

In [13]:
# dropping the uncleaned text column "text"
df = df.drop(['text'], axis=1)

In [14]:
# renaming the column
df.rename(columns = {'clean_text':'text'}, inplace = True)
df.head()

Index  Class  \
0     99      1   
1    185      1   
2   1574      0   
3    523      0   
4   1548      0   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              text  
0  We stayed at the Sheraton by Navy Pier the first weekend of November and were very pleased with the accommodations and the service We were upgraded to club level rooms and stayed in and adjoining The view from both rooms was spectacular They also left a plate of cookies and treats in the kids room upon check in which made us all feel very special The hotel is central to both Navy Pier and Michigan Ave so we walked trolleyed and cabbed all around the area We ate the breakfast buffet both mornings and thought it was pretty good The eggs were a little runny but they happily made a different batch for us at the omelet station Our six year old ate free and our two eleven year olds were The rooms were clean the concierge reception staff were both friendly and helpful we will definitely visit...  
1  Got a great deal through Hotwire for a night but if I ever return to Chicago I would definitely stay again even at full price We really liked the rooms muted modern decor we had John Lennon and Bono for company Comfortable bed mini coffe maker plenty storage space and a lovely clean bathroom stocked full of Aveda products If you re a product junkie tip the maid and request some extra bottles and you will leave with lots of goodies We liked the location and had a great view down N Michigan Ave found the staff to be really friendly and relaxed The bar area is very trendy I liked the chilled atmosphere which surprised me cause I normally find these types of bars a bit too pretentious for my liking the lack of memorabilia didn t bother me but I was more interested in the art deco features ...  
2  My wife and I were excited to visit and shop at Chicago s Magnificent Mile on Michigan Ave We chose the Intercontinental Hotel for it s reputation and location Man oh man were we disappointed The check in was smooth enough even though the hotel was busy and all seemed well We got ready to go out shopping and went to visit the concierge in the lobby to ask about suggestions and bargains There were six people in line for her and we had to wait That was no problem since we weren t on a timetable When we finally got to her and asked about shopping on the Miracle Mile she just pointed to the door and said It s right out the door I asked her for suggestions of the best stores for bargains and she rolled her eyes at me saying there s no such thing I couldn t believe her abruptness We left the...  
3                                                                                                                                                                                                                                                                                                                                                                                                                     I must say this hotel is absolutely beautiful My wife and I were blown away by the penthouse fitness center and pool all the way up on the nd floor Very well designed rooms with great linens and pillows The bathrooms were a nice change of pace from the boring ho

In [15]:
import spacy
#loading the english language small model of spacy
en = spacy.load('en_core_web_sm')
sw_spacy = en.Defaults.stop_words
print(sw_spacy)

/usr/local/lib/python3.8/dist-packages/torch/cuda/__init__.py:497: UserWarning: Can't initialize NVML
  warnings.warn("Can't initialize NVML")


{'on', 'few', 'thence', 'such', 'before', 'many', 'his', 'out', 'indeed', 'make', 'nowhere', 'off', 'own', 'whatever', 'me', 'perhaps', 'thereby', 'neither', 'how', 'thru', 'do', 'to', 'formerly', 'sometime', 'hers', 'while', 'with', 'is', 'more', 'otherwise', 'behind', 'latter', 'eight', 'up', 'using', 'an', 'name', 'anywhere', 'your', 'mostly', 'am', 'these', 'beside', 'whenever', "'m", 'again', 'always', 'whoever', 'might', 'every', 'take', 'something', 'whence', 'although', 'one', 'everything', 'be', 'forty', 'never', 'at', 'thereafter', 'under', 'cannot', 'when', 'per', 'below', '’m', 'around', 'twenty', 'now', 'less', 'thus', 'hereupon', 'next', 'seeming', 'really', 'except', 'their', '’d', 'became', 'because', 'her', 'you', 'himself', 'i', 'latterly', 'herein', 'bottom', 'in', 'four', 'please', 'which', 'them', 'whole', 'onto', 'part', 'hundred', 'last', 'hereafter', 'regarding', 'however', '’ll', 'used', 'get', 'we', 'anything', 'still', 'us', 'after', 'yourself', 'can', 'third

In [16]:
import nltk
from nltk.corpus import stopwords
sw_nltk = stopwords.words('english')
print(sw_nltk)

['i', 'me', 'my', 'myself', 'we', 'our', 'ours', 'ourselves', 'you', "you're", "you've", "you'll", "you'd", 'your', 'yours', 'yourself', 'yourselves', 'he', 'him', 'his', 'himself', 'she', "she's", 'her', 'hers', 'herself', 'it', "it's", 'its', 'itself', 'they', 'them', 'their', 'theirs', 'themselves', 'what', 'which', 'who', 'whom', 'this', 'that', "that'll", 'these', 'those', 'am', 'is', 'are', 'was', 'were', 'be', 'been', 'being', 'have', 'has', 'had', 'having', 'do', 'does', 'did', 'doing', 'a', 'an', 'the', 'and', 'but', 'if', 'or', 'because', 'as', 'until', 'while', 'of', 'at', 'by', 'for', 'with', 'about', 'against', 'between', 'into', 'through', 'during', 'before', 'after', 'above', 'below', 'to', 'from', 'up', 'down', 'in', 'out', 'on', 'off', 'over', 'under', 'again', 'further', 'then', 'once', 'here', 'there', 'when', 'where', 'why', 'how', 'all', 'any', 'both', 'each', 'few', 'more', 'most', 'other', 'some', 'such', 'no', 'nor', 'not', 'only', 'own', 'same', 'so', 'than', '

In [17]:
sw_nltk.extend(['th','stay'])
print(len(sw_nltk))

181


In [18]:
# preprocessing the data (removing stopwords, convert the sentence to lower case, and also applying lemmatization to get meaningful words )
from nltk.corpus import stopwords
from nltk.stem import WordNetLemmatizer, PorterStemmer
#stop_words = set(stopwords.words('english'))
ps=PorterStemmer()
wnl=WordNetLemmatizer()
def clean_lemma(review):
    review = review.lower()
    review = review.split()
    review = [wnl.lemmatize(word) for word in review if word not in sw_nltk]
    review = [wnl.lemmatize(word) for word in review if word not in sw_spacy]
    review = ' '.join(review)
    return review

In [19]:
# using lemmatization
df['text']=df['text'].apply(clean_lemma)
df.head()

Index  Class  \
0     99      1   
1    185      1   
2   1574      0   
3    523      0   
4   1548      0   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              text  
0                                                                                                                                                                                                                                                                                                                               stayed sheraton navy pier weekend november pleased accommodation service upgraded club level room stayed adjoining view room spectacular left plate cooky treat kid room check u feel special hotel central navy pier michigan ave walked trolleyed cabbed area ate breakfast buffet morning thought pretty good egg little runny happily different batch u omelet station year old ate free year old room clean concierge reception staff friendly helpful definitely visit sheraton chicago time  
1                                                                                                                   got great deal hotwire night return chicago definitely price liked room muted modern decor john lennon bono company comfortable bed mini coffe maker plenty storage space lovely clean bathroom stocked aveda product product junkie tip maid request extra bottle leave lot goody liked location great view n michigan ave found staff friendly relaxed bar area trendy liked chilled atmosphere surprised cause normally find type bar bit pretentious liking lack memorabilia bother interested art deco feature building fault speaker room crackling bit oh boyfriend think propositioned friendly lady looking business bar think place love hate like laid contemporary hotel intimidating factor place  
2  wife excited visit shop chicago magnificent mile michigan ave chose intercontinental hotel reputation location man oh man disappointed check smooth hotel busy got ready shopping went visit concierge lobby ask suggestion bargain people line wait problem timetable finally got asked shopping miracle mile pointed door said right door asked suggestion best store bargain rolled eye saying thing believe abruptness left concierge stand wanting return shopping went fun returned room wanted look restaurant night ask concierge find hotel directory room called housekeeping department told directory office asked check vacant room said hotel vacant room hung thought look restaurant yellow page guess guessed phone book room figured drink bar hotel themed lounge ask bartender suggestion went bar custo...  
3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        hotel absolutely beautiful wife blown away penthouse fitness center pool way nd floor designed room

In [20]:
df1 = df.sort_values(by=['Index'], ascending=True)
df1.head()

Index  Class  \
678      1      1   
774      2      1   
441      3      1   
82       4      1   
852      5      1   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        text  
678                                                                                                                                                                                                                                                            stayed night getaway family thursday triple aaa rate steal floor room complete plasma tv bose stereo voss evian water gorgeous bathroom concierge helpful beat location flaw breakfast pricey service slow table restaurant food good worth wait return heartbeat gem chicago  
774                                                                                                                                                                                                                                                                                                                                 triple rate upgrade view room le included breakfast voucher great view river lake wrigley bldg tribune bldg major restaurant shopping sightseeing attraction walking distance large room comfortable bed  
441  come little late finally catching review past month dear friend stayed hyatt regency late october night visiting friend husband town hotel perfect imo easy check check lovely clean comfortable room great view city know area pretty convenient downtown chicago attraction dinner went clubing friend division st problem getting cab forth hyatt public transportation right near bother needed cab hotel parking usual chicago expensive able car quickly problem best got rate hotwire downright steal area chicago quality hotel  
82                                                                                                                       omni chicago delivers spaciousness room helpful staff prized location michigan avenue address chicago requires high level quality omni delivers check group people minute staff plentiful recommendation dining event room largest find price range chicago standard room separate living area work desk fitness center free weight weight machine row cardio equipment shared room feel cramped way great property  
852                                                                                                                                                                                                                                                                                                                                asked high floor away elevator got room pleasantly decorated functional clean need lot service pleasant prompt fitness center equipped working order great location end michigan avenue shopping district

In [21]:
df1.reset_index(drop=True, inplace=True)

In [22]:
df1.head()

Index  Class  \
0      1      1   
1      2      1   
2      3      1   
3      4      1   
4      5      1   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      text  
0                                                                                                                                                                                                                                                            stayed night getaway family thursday triple aaa rate steal floor room complete plasma tv bose stereo voss evian water gorgeous bathroom concierge helpful beat location flaw breakfast pricey service slow table restaurant food good worth wait return heartbeat gem chicago  
1                                                                                                                                                                                                                                                                                                                                 triple rate upgrade view room le included breakfast voucher great view river lake wrigley bldg tribune bldg major restaurant shopping sightseeing attraction walking distance large room comfortable bed  
2  come little late finally catching review past month dear friend stayed hyatt regency late october night visiting friend husband town hotel perfect imo easy check check lovely clean comfortable room great view city know area pretty convenient downtown chicago attraction dinner went clubing friend division st problem getting cab forth hyatt public transportation right near bother needed cab hotel parking usual chicago expensive able car quickly problem best got rate hotwire downright steal area chicago quality hotel  
3                                                                                                                      omni chicago delivers spaciousness room helpful staff prized location michigan avenue address chicago requires high level quality omni delivers check group people minute staff plentiful recommendation dining event room largest find price range chicago standard room separate living area work desk fitness center free weight weight machine row cardio equipment shared room feel cramped way great property  
4                                                                                                                                                                                                                                                                                                                                asked high floor away elevator got room pleasantly decorated functional clean need lot service pleasant prompt fitness center equipped working order great location end michigan avenue shopping district

### FEATURE EXTRACTION (Bag of words)

In [23]:
x = df1['text']
y = df1['Class']

In [24]:
y

0       1
1       1
2       1
3       1
4       1
       ..
1595    0
1596    0
1597    0
1598    0
1599    0
Name: Class, Length: 1600, dtype: int64

In [25]:
from sklearn.feature_extraction.text import TfidfVectorizer, CountVectorizer 
from sklearn.feature_extraction.text import TfidfVectorizer
cv=CountVectorizer()
corpii=cv.fit_transform(x) 

In [26]:
feature_names = cv.get_feature_names()
#feature_names

/usr/local/lib/python3.8/dist-packages/sklearn/utils/deprecation.py:87: FutureWarning: Function get_feature_names is deprecated; get_feature_names is deprecated in 1.0 and will be removed in 1.2. Please use get_feature_names_out instead.
  warnings.warn(msg, category=FutureWarning)


In [27]:
len(feature_names)

7905

In [28]:
u1 = pd.DataFrame(corpii.toarray(), columns=feature_names)
u1.head()

aaa  aaahed  aback  abassador  abd  ability  able  abound  abrupt  \
0    1       0      0          0    0        0     0       0       0   
1    0       0      0          0    0        0     0       0       0   
2    0       0      0          0    0        0     1       0       0   
3    0       0      0          0    0        0     0       0       0   
4    0       0      0          0    0        0     0       0       0   

   abruptly  ...  yucky  yummo  yummy  yunan  yup  zagat  zest  zipped  zone  \
0         0  ...      0      0      0      0    0      0     0       0     0   
1         0  ...      0      0      0      0    0      0     0       0     0   
2         0  ...      0      0      0      0    0      0     0       0     0   
3         0  ...      0      0      0      0    0      0     0       0     0   
4         0  ...      0      0      0      0    0      0     0       0     0   

   zoo  
0    0  
1    0  
2    0  
3    0  
4    0  

[5 rows x 7905 columns]

### FEATURE SELECTION (CHI- SQUARE)

In [29]:
from sklearn.feature_selection import SelectKBest
from sklearn.feature_selection import chi2
# highest chi-squared statistics are selected
chi2_features = SelectKBest(chi2, k = 2000)
X_kbest_features = chi2_features.fit_transform(u1, y)

In [30]:
chi_support = chi2_features.get_support()

In [31]:
chi_feature = u1.loc[:,chi_support].columns.tolist()

In [32]:
X_kbest_features

array([[0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       ...,
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0]])

In [33]:
u3 = pd.DataFrame(X_kbest_features, columns = chi_feature)
u3.head()

abassador  absent  absolute  absurd  accept  accepted  accessible  \
0          0       0         0       0       0         0           0   
1          0       0         0       0       0         0           0   
2          0       0         0       0       0         0           0   
3          0       0         0       0       0         0           0   
4          0       0         0       0       0         0           0   

   accomidations  accommodated  accommodation  ...  yep  yes  yesterday  \
0              0             0              0  ...    0    0          0   
1              0             0              0  ...    0    0          0   
2              0             0              0  ...    0    0          0   
3              0             0              0  ...    0    0          0   
4              0             0              0  ...    0    0          0   

   yielded  yoga  young  youve  yr  yummy  zone  
0        0     0      0      0   0      0     0  
1        0     0      0      0   0      0     0  
2        0     0      0      0   0      0     0  
3        0     0      0      0   0      0     0  
4        0     0      0      0   0      0     0  

[5 rows x 2000 columns]

In [34]:
X1 = pd.concat([u3, y], axis=1)
X1.head()

abassador  absent  absolute  absurd  accept  accepted  accessible  \
0          0       0         0       0       0         0           0   
1          0       0         0       0       0         0           0   
2          0       0         0       0       0         0           0   
3          0       0         0       0       0         0           0   
4          0       0         0       0       0         0           0   

   accomidations  accommodated  accommodation  ...  yes  yesterday  yielded  \
0              0             0              0  ...    0          0        0   
1              0             0              0  ...    0          0        0   
2              0             0              0  ...    0          0        0   
3              0             0              0  ...    0          0        0   
4              0             0              0  ...    0          0        0   

   yoga  young  youve  yr  yummy  zone  Class  
0     0      0      0   0      0     0      1  
1     0      0      0   0      0     0      1  
2     0      0      0   0      0     0      1  
3     0      0      0   0      0     0      1  
4     0      0      0   0      0     0      1  

[5 rows x 2001 columns]

In [35]:
u3.head()

abassador  absent  absolute  absurd  accept  accepted  accessible  \
0          0       0         0       0       0         0           0   
1          0       0         0       0       0         0           0   
2          0       0         0       0       0         0           0   
3          0       0         0       0       0         0           0   
4          0       0         0       0       0         0           0   

   accomidations  accommodated  accommodation  ...  yep  yes  yesterday  \
0              0             0              0  ...    0    0          0   
1              0             0              0  ...    0    0          0   
2              0             0              0  ...    0    0          0   
3              0             0              0  ...    0    0          0   
4              0             0              0  ...    0    0          0   

   yielded  yoga  young  youve  yr  yummy  zone  
0        0     0      0      0   0      0     0  
1        0     0      0      0   0      0     0  
2        0     0      0      0   0      0     0  
3        0     0      0      0   0      0     0  
4        0     0      0      0   0      0     0  

[5 rows x 2000 columns]

In [36]:
df1['Class'].value_counts()

1    800
0    800
Name: Class, dtype: int64

In [37]:
from sklearn.model_selection import train_test_split
from sklearn import tree
from sklearn.metrics import classification_report,accuracy_score
from sklearn import preprocessing
from sklearn.linear_model import LogisticRegression
from sklearn.linear_model import LinearRegression
from sklearn.svm import SVC
from sklearn.svm import LinearSVC
from sklearn import metrics

from sklearn.ensemble import RandomForestClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn.naive_bayes import MultinomialNB
from sklearn.svm import LinearSVC
from xgboost import XGBClassifier
from xgboost import XGBRegressor
from sklearn.metrics import confusion_matrix, recall_score, precision_score,f1_score
import seaborn as sns

In [38]:
X=u3
Y=y

## Model Building

### CROSS VALIDATION

In [39]:
from sklearn.model_selection import KFold
from sklearn.model_selection import StratifiedKFold
from sklearn.model_selection import cross_val_score
cv = KFold(n_splits=10, random_state=7, shuffle=True)
kf = KFold(n_splits=10)

In [40]:
from sklearn.svm import LinearSVC
models = LinearSVC()
folds = StratifiedKFold(n_splits=10, shuffle=True, random_state=7)
y_pred = []
y_exp = []

for train_index, test_index in folds.split(X, Y):
    x_train, x_test = X.iloc[train_index], X.iloc[test_index]
    y_train, y_test = Y[train_index], Y[test_index]
    models.fit(x_train, y_train)
    # store result from classification
    y_pred.extend(models.predict(x_test))
    # store expected result for this specific fold
    y_exp.extend(y_test)

print("Accuracy: ",  accuracy_score(y_exp, y_pred))
print(classification_report(y_exp, y_pred))
print("Confusion matrix: ", confusion_matrix(y_exp, y_pred))

Accuracy:  0.870625
              precision    recall  f1-score   support

           0       0.86      0.89      0.87       800
           1       0.88      0.85      0.87       800

    accuracy                           0.87      1600
   macro avg       0.87      0.87      0.87      1600
weighted avg       0.87      0.87      0.87      1600

Confusion matrix:  [[710  90]
 [117 683]]


In [41]:
from sklearn.linear_model import PassiveAggressiveClassifier
pag=PassiveAggressiveClassifier(max_iter=1000, random_state=7, tol=1e-3)
folds = StratifiedKFold(n_splits=10, shuffle=True, random_state=7)
y_pred = []
y_exp = []

for train_index, test_index in folds.split(X, Y):
    x_train, x_test = X.iloc[train_index], X.iloc[test_index]
    y_train, y_test = Y[train_index], Y[test_index]
    pag.fit(x_train, y_train)
    # store result from classification
    y_pred.extend(pag.predict(x_test))
    # store expected result for this specific fold
    y_exp.extend(y_test)

print("Accuracy: ",  accuracy_score(y_exp, y_pred))
print(classification_report(y_exp, y_pred))
print("Confusion matrix: ", confusion_matrix(y_exp, y_pred))

Accuracy:  0.860625
              precision    recall  f1-score   support

           0       0.85      0.87      0.86       800
           1       0.87      0.85      0.86       800

    accuracy                           0.86      1600
   macro avg       0.86      0.86      0.86      1600
weighted avg       0.86      0.86      0.86      1600

Confusion matrix:  [[699 101]
 [122 678]]


In [42]:
from sklearn.linear_model import LogisticRegression
lr = LogisticRegression(max_iter=10000) 
folds = StratifiedKFold(n_splits=10, shuffle=True, random_state=7)
y_pred = []
y_exp = []

for train_index, test_index in folds.split(X, Y):
    x_train, x_test = X.iloc[train_index], X.iloc[test_index]
    y_train, y_test = Y[train_index], Y[test_index]
    lr.fit(x_train, y_train)
    # store result from classification
    y_pred.extend(lr.predict(x_test))
    # store expected result for this specific fold
    y_exp.extend(y_test)

print("Accuracy: ",  accuracy_score(y_exp, y_pred))
print(classification_report(y_exp, y_pred))
print("Confusion matrix: ", confusion_matrix(y_exp, y_pred))

Accuracy:  0.8825
              precision    recall  f1-score   support

           0       0.87      0.90      0.88       800
           1       0.90      0.86      0.88       800

    accuracy                           0.88      1600
   macro avg       0.88      0.88      0.88      1600
weighted avg       0.88      0.88      0.88      1600

Confusion matrix:  [[722  78]
 [110 690]]


In [43]:
from sklearn.ensemble import RandomForestClassifier, GradientBoostingClassifier, AdaBoostClassifier
rf = RandomForestClassifier(n_estimators=300)
folds = StratifiedKFold(n_splits=10, shuffle=True, random_state=7)
y_pred = []
y_exp = []

for train_index, test_index in folds.split(X, Y):
    x_train, x_test = X.iloc[train_index], X.iloc[test_index]
    y_train, y_test = Y[train_index], Y[test_index]
    rf.fit(x_train, y_train)
    # store result from classification
    y_pred.extend(rf.predict(x_test))
    # store expected result for this specific fold
    y_exp.extend(y_test)

print("Accuracy: ",  accuracy_score(y_exp, y_pred))
print(classification_report(y_exp, y_pred))
print("Confusion matrix: ", confusion_matrix(y_exp, y_pred))

Accuracy:  0.876875
              precision    recall  f1-score   support

           0       0.86      0.90      0.88       800
           1       0.89      0.86      0.87       800

    accuracy                           0.88      1600
   macro avg       0.88      0.88      0.88      1600
weighted avg       0.88      0.88      0.88      1600

Confusion matrix:  [[717  83]
 [114 686]]


In [44]:
gbc = GradientBoostingClassifier(n_estimators=300)
folds = StratifiedKFold(n_splits=10, shuffle=True, random_state=7)
y_pred = []
y_exp = []

for train_index, test_index in folds.split(X, Y):
    x_train, x_test = X.iloc[train_index], X.iloc[test_index]
    y_train, y_test = Y[train_index], Y[test_index]
    gbc.fit(x_train, y_train)
    # store result from classification
    y_pred.extend(gbc.predict(x_test))
    # store expected result for this specific fold
    y_exp.extend(y_test)

print("Accuracy: ",  accuracy_score(y_exp, y_pred))
print(classification_report(y_exp, y_pred))
print("Confusion matrix: ", confusion_matrix(y_exp, y_pred))

Accuracy:  0.85125
              precision    recall  f1-score   support

           0       0.85      0.85      0.85       800
           1       0.85      0.85      0.85       800

    accuracy                           0.85      1600
   macro avg       0.85      0.85      0.85      1600
weighted avg       0.85      0.85      0.85      1600

Confusion matrix:  [[678 122]
 [116 684]]


In [45]:
abc = AdaBoostClassifier(n_estimators=300)
folds = StratifiedKFold(n_splits=10, shuffle=True, random_state=7)
y_pred = []
y_exp = []

for train_index, test_index in folds.split(X, Y):
    x_train, x_test = X.iloc[train_index], X.iloc[test_index]
    y_train, y_test = Y[train_index], Y[test_index]
    abc.fit(x_train, y_train)
    # store result from classification
    y_pred.extend(abc.predict(x_test))
    # store expected result for this specific fold
    y_exp.extend(y_test)

print("Accuracy: ",  accuracy_score(y_exp, y_pred))
print(classification_report(y_exp, y_pred))
print("Confusion matrix: ", confusion_matrix(y_exp, y_pred))

Accuracy:  0.828125
              precision    recall  f1-score   support

           0       0.83      0.83      0.83       800
           1       0.83      0.82      0.83       800

    accuracy                           0.83      1600
   macro avg       0.83      0.83      0.83      1600
weighted avg       0.83      0.83      0.83      1600

Confusion matrix:  [[665 135]
 [140 660]]


In [46]:
import xgboost as xgb
xgbc = xgb.XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,
              colsample_bynode=1, colsample_bytree=1, 
              gamma=0,  importance_type='gain',
              interaction_constraints='', learning_rate=0.300000012,
              max_delta_step=0, max_depth=6, min_child_weight=1, 
             n_estimators=200, n_jobs=16,
              num_parallel_tree=1, random_state=0,
              reg_alpha=0, reg_lambda=1, subsample=1,
              tree_method='exact', use_label_encoder=False
              )
folds = StratifiedKFold(n_splits=10, shuffle=True, random_state=7)
y_pred = []
y_exp = []

for train_index, test_index in folds.split(X, Y):
    x_train, x_test = X.iloc[train_index], X.iloc[test_index]
    y_train, y_test = Y[train_index], Y[test_index]
    xgbc.fit(x_train, y_train)
    # store result from classification
    y_pred.extend(xgbc.predict(x_test))
    # store expected result for this specific fold
    y_exp.extend(y_test)

print("Accuracy: ",  accuracy_score(y_exp, y_pred))
print(classification_report(y_exp, y_pred))
print("Confusion matrix: ", confusion_matrix(y_exp, y_pred))

Accuracy:  0.8475
              precision    recall  f1-score   support

           0       0.84      0.85      0.85       800
           1       0.85      0.84      0.85       800

    accuracy                           0.85      1600
   macro avg       0.85      0.85      0.85      1600
weighted avg       0.85      0.85      0.85      1600

Confusion matrix:  [[681 119]
 [125 675]]


In [47]:
import lightgbm as lgb
lgb = lgb.LGBMClassifier(boosting_type='gbdt', class_weight=None, colsample_bytree=1.0,
               importance_type='split', learning_rate=0.1, max_depth=-1,
               min_child_samples=20, min_child_weight=0.001, min_split_gain=0.0,
               n_estimators=400, n_jobs=-1, num_leaves=31, objective=None,
               random_state=None, reg_alpha=0.0, reg_lambda=0.0, silent=True,
               subsample=1.0, subsample_for_bin=200000, subsample_freq=0)
folds = StratifiedKFold(n_splits=10, shuffle=True, random_state=7)
y_pred = []
y_exp = []

for train_index, test_index in folds.split(X, Y):
    x_train, x_test = X.iloc[train_index], X.iloc[test_index]
    y_train, y_test = Y[train_index], Y[test_index]
    lgb.fit(x_train, y_train)
    # store result from classification
    y_pred.extend(lgb.predict(x_test))
    # store expected result for this specific fold
    y_exp.extend(y_test)

print("Accuracy: ",  accuracy_score(y_exp, y_pred))
print(classification_report(y_exp, y_pred))
print("Confusion matrix: ", confusion_matrix(y_exp, y_pred))

Accuracy:  0.84625
              precision    recall  f1-score   support

           0       0.84      0.85      0.85       800
           1       0.85      0.84      0.85       800

    accuracy                           0.85      1600
   macro avg       0.85      0.85      0.85      1600
weighted avg       0.85      0.85      0.85      1600

Confusion matrix:  [[683 117]
 [129 671]]


### Train test split

In [48]:
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(u3, y, test_size=0.2, random_state=7, stratify=y)

In [49]:
from sklearn.linear_model import PassiveAggressiveClassifier
pa=PassiveAggressiveClassifier(max_iter=1000, random_state=7, tol=1e-3)
pa.fit(x_train, y_train)
y_pred=pa.predict(x_test)
from sklearn.metrics import confusion_matrix, classification_report, accuracy_score
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))
print('accuracy_score: ', accuracy_score(y_test, y_pred))

[[138  22]
 [ 26 134]]
              precision    recall  f1-score   support

           0       0.84      0.86      0.85       160
           1       0.86      0.84      0.85       160

    accuracy                           0.85       320
   macro avg       0.85      0.85      0.85       320
weighted avg       0.85      0.85      0.85       320

accuracy_score:  0.85


In [50]:
from sklearn.svm import LinearSVC
lsvm = LinearSVC()
lsvm.fit(x_train, y_train)
y_pred=lsvm.predict(x_test)
from sklearn.metrics import confusion_matrix, classification_report, accuracy_score
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))
print('accuracy_score: ', accuracy_score(y_test, y_pred))

[[136  24]
 [ 29 131]]
              precision    recall  f1-score   support

           0       0.82      0.85      0.84       160
           1       0.85      0.82      0.83       160

    accuracy                           0.83       320
   macro avg       0.83      0.83      0.83       320
weighted avg       0.83      0.83      0.83       320

accuracy_score:  0.834375


In [51]:
rf = RandomForestClassifier(n_estimators=300)
rf.fit(x_train, y_train)
y_pred=rf.predict(x_test)
from sklearn.metrics import confusion_matrix, classification_report, accuracy_score
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))
print('accuracy_score: ', accuracy_score(y_test, y_pred))

[[146  14]
 [ 24 136]]
              precision    recall  f1-score   support

           0       0.86      0.91      0.88       160
           1       0.91      0.85      0.88       160

    accuracy                           0.88       320
   macro avg       0.88      0.88      0.88       320
weighted avg       0.88      0.88      0.88       320

accuracy_score:  0.88125


In [52]:
xgbc = xgb.XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,
              colsample_bynode=1, colsample_bytree=1, 
              gamma=0,  importance_type='gain',
              interaction_constraints='', learning_rate=0.300000012,
              max_delta_step=0, max_depth=6, min_child_weight=1, 
             n_estimators=200, n_jobs=16,
              num_parallel_tree=1, random_state=0,
              reg_alpha=0, reg_lambda=1, subsample=1,
              tree_method='exact', use_label_encoder=False
              )
xgbc.fit(x_train, y_train)
y_pred=xgbc.predict(x_test)
from sklearn.metrics import confusion_matrix, classification_report, accuracy_score
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))
print('accuracy_score: ', accuracy_score(y_test, y_pred))

[[140  20]
 [ 20 140]]
              precision    recall  f1-score   support

           0       0.88      0.88      0.88       160
           1       0.88      0.88      0.88       160

    accuracy                           0.88       320
   macro avg       0.88      0.88      0.88       320
weighted avg       0.88      0.88      0.88       320

accuracy_score:  0.875


In [53]:
log = LogisticRegression(max_iter=10000) 
log.fit(x_train, y_train)
y_pred=log.predict(x_test)
from sklearn.metrics import confusion_matrix, classification_report, accuracy_score
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))
print('accuracy_score: ', accuracy_score(y_test, y_pred))

[[143  17]
 [ 20 140]]
              precision    recall  f1-score   support

           0       0.88      0.89      0.89       160
           1       0.89      0.88      0.88       160

    accuracy                           0.88       320
   macro avg       0.88      0.88      0.88       320
weighted avg       0.88      0.88      0.88       320

accuracy_score:  0.884375


In [54]:
gbc = GradientBoostingClassifier(n_estimators=300)
gbc.fit(x_train, y_train)
y_pred=gbc.predict(x_test)
from sklearn.metrics import confusion_matrix, classification_report, accuracy_score
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))
print('accuracy_score: ', accuracy_score(y_test, y_pred))

[[141  19]
 [ 25 135]]
              precision    recall  f1-score   support

           0       0.85      0.88      0.87       160
           1       0.88      0.84      0.86       160

    accuracy                           0.86       320
   macro avg       0.86      0.86      0.86       320
weighted avg       0.86      0.86      0.86       320

accuracy_score:  0.8625


In [55]:
abc = AdaBoostClassifier(n_estimators=300)
abc.fit(x_train, y_train)
y_pred=abc.predict(x_test)
from sklearn.metrics import confusion_matrix, classification_report, accuracy_score
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))
print('accuracy_score: ', accuracy_score(y_test, y_pred))

[[136  24]
 [ 28 132]]
              precision    recall  f1-score   support

           0       0.83      0.85      0.84       160
           1       0.85      0.82      0.84       160

    accuracy                           0.84       320
   macro avg       0.84      0.84      0.84       320
weighted avg       0.84      0.84      0.84       320

accuracy_score:  0.8375


In [56]:
import lightgbm as lgb
lgb = lgb.LGBMClassifier(boosting_type='gbdt', class_weight=None, colsample_bytree=1.0,
               importance_type='split', learning_rate=0.1, max_depth=-1,
               min_child_samples=20, min_child_weight=0.001, min_split_gain=0.0,
               n_estimators=400, n_jobs=-1, num_leaves=31, objective=None,
               random_state=None, reg_alpha=0.0, reg_lambda=0.0, silent=True,
               subsample=1.0, subsample_for_bin=200000, subsample_freq=0)
lgb.fit(x_train, y_train)
y_pred=lgb.predict(x_test)
from sklearn.metrics import confusion_matrix, classification_report, accuracy_score
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))
print('accuracy_score: ', accuracy_score(y_test, y_pred))

[[143  17]
 [ 25 135]]
              precision    recall  f1-score   support

           0       0.85      0.89      0.87       160
           1       0.89      0.84      0.87       160

    accuracy                           0.87       320
   macro avg       0.87      0.87      0.87       320
weighted avg       0.87      0.87      0.87       320

accuracy_score:  0.86875


### Using logistic regression with 10-fold cross validation, we achieve 88% accuracy.